File: frontend/src/lib/mock.ts
import type {
  Citation,
  GenerationParams,
  OutputTypeId,
} from "./types";

function sleep(ms: number) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

export async function generatePlan(
  sourceText: string,
  fileNames: string[],
  links: string[],
  outputs: { id: OutputTypeId; params: GenerationParams }[]
): Promise<{ plan: string; citations: Citation[] }> {
  await sleep(600);

  const citations: Citation[] = [];
  let citIdx = 1;

  if (sourceText.trim()) {
    citations.push({
      id: `cit-${citIdx++}`,
      label: "Raw Telemetry & Advisory Input",
      kind: "text",
    });
  }

  for (const f of fileNames) {
    citations.push({
      id: `cit-${citIdx++}`,
      label: f,
      kind: "file",
    });
  }

  for (const l of links) {
    try {
      const url = new URL(l);
      citations.push({
        id: `cit-${citIdx++}`,
        label: url.hostname + url.pathname,
        kind: "link",
      });
    } catch {
      citations.push({
        id: `cit-${citIdx++}`,
        label: l,
        kind: "link",
      });
    }
  }

  const plan = `### Grounding Transformation Plan

**Context Analysis:**
- **Source Modalities:** ${fileNames.length ? `${fileNames.length} attached file(s), ` : ""}${links.length ? `${links.length} external intelligence link(s), ` : ""}direct narrative text (${sourceText.trim().length} chars).
- **Target Deliverables:** ${outputs.map((o) => o.id).join(", ")}

**Execution Directives:**
1. **Fact Locking:** Canonical anchor extracted for threat actors, CVEs, IOCs, and timelines to prevent multi-agent hallucination.
2. **Audience Alignment:** Enforcing tone (${outputs[0]?.params.tone || "Authoritative"}) and detail level (${outputs[0]?.params.detail || "Standard"}).
3. **Citation Tagging:** Every technical claim will reference verified source entities.

*Feel free to edit this blueprint before proceeding with generation.*`;

  return { plan, citations };
}

export async function generateDeliverable(
  id: OutputTypeId,
  sourceText: string,
  params: GenerationParams
): Promise<string> {
  await sleep(700);

  const previewSnippet = sourceText.trim()
    ? sourceText.trim().slice(0, 150) + "..."
    : "Critical system event analysis";
  const dateStr = new Date().toISOString().split("T")[0];

  switch (id) {
    case "advisory":
      return `# TECHNICAL THREAT ADVISORY: TA-2026-${Math.floor(1000 + Math.random() * 9000)}
**Published Date:** ${dateStr}  
**Severity:** CRITICAL (CVSS 9.8)  
**Target Audience:** ${params.targetAudience || "SOC Analysts, CERT Teams, CISO Staff"}  
**Language:** ${params.language}

---

## 1. Executive Summary
An ongoing cyber campaign has been detected exploiting vulnerabilities in perimeter infrastructure. Initial telemetry suggests zero-day vector chaining resulting in unauthorized privilege escalation.

> **Source Summary:**  
> ${previewSnippet}

## 2. Technical Analysis & Attack Vector
- **Initial Access:** Weaponized payloads bypassing web application firewalls.
- **Execution & Persistence:** Process hollowing and scheduled cron modification.
- **Lateral Movement:** Pass-the-hash techniques across internal subnets.

## 3. Indicators of Compromise (IOCs)
| Type | Indicator | Context / Notes |
| :--- | :--- | :--- |
| **SHA-256** | \`e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\` | Stager binary |
| **IPv4** | \`198.51.100.42\` | C2 Infrastructure |
| **Domain** | \`telemetry-sync-auth[.]net\` | Exfiltration Endpoint |

## 4. Recommended Mitigations
1. **Immediate Quarantine:** Block inbound traffic from listed C2 IPs at firewall boundaries.
2. **Patching:** Apply vendor security updates immediately.
3. **Credential Invalidation:** Force global password resets for compromised administrator accounts.`;

    case "exec_summary":
      return `# Executive Brief: Cybersecurity Incident & Risk Assessment
**Date:** ${dateStr}  
**Classification:** STRICTLY CONFIDENTIAL  
**Prepared For:** Executive Board & C-Suite  

---

### Situation
A sophisticated attack campaign has targeted key digital infrastructure components. Our detection systems have quarantined the threat, preventing enterprise data compromise.

### Complication & Business Risk
- **Operational Exposure:** 3 critical internal services required temporary isolation for forensics.
- **Regulatory Impact:** Mandatory 72-hour notification threshold actively monitored.
- **Estimated Remediation Downtime:** Under 2 hours.

### Strategic Recommendations & ROI
1. **Approve Enhanced EDR Rollout:** Expedite autonomous endpoint detection to lower dwell time by 60%.
2. **Resource Allocation:** Empower the incident response team to perform deep forensic sweeps across secondary clusters.
3. **Customer Assurance:** Maintain transparent communication with key enterprise partners.`;

    case "incident_report":
      return `# Incident Triage & Forensic Report
**Incident ID:** INC-2026-9812  
**Status:** CONTAINED / UNDER TRIAGE  
**Severity Level:** Tier 1 High  

---

## Timeline of Events (UTC)
- **08:14:22** - Initial anomalous ingress traffic flagged from foreign subnet.
- **08:21:05** - Privilege escalation alert triggered on core API gateway.
- **08:35:00** - SOC initiated perimeter containment and IP blocklist push.
- **09:10:14** - Host isolation completed; zero persistence mechanisms found on database layer.

## Root Cause Analysis
The attack exploited an unpatched deserialization vulnerability in public-facing ingestion endpoints.

## Corrective Actions
- [x] Ingress point patched and recompiled
- [x] Network segmentation policy re-enforced
- [ ] 14-day continuous forensic logging activated`;

    case "social_thread":
      return `🧵 **1/5 THREAT ALERT: Critical Advisory Breakdown**

A high-severity exploitation campaign is targeting enterprise systems. Here is everything your security team needs to know in 60 seconds 👇

---

**2/5 🔍 The Attack Vector**
Threat actors are leveraging deserialization zero-days to achieve unauthenticated remote code execution. Perimeter logging shows active scanning in the wild.

---

**3/5 🛡️ Key Indicators (IOCs)**
- Block C2 IP: \`198.51.100.42\`
- Search for hashes matching stager binaries
- Review suspicious API gateway parent processes

---

**4/5 ⚡ Immediate Action Items**
1️⃣ Segment management interfaces  
2️⃣ Review external access audit logs  
3️⃣ Patch immediately upon vendor advisory availability  

---

**5/5 🔗 Stay Safe**
Full technical advisory and IOC hashes available via official channels. Retweet to inform the community. #CyberSecurity #ThreatIntel #InfoSec`;

    case "linkedin_post":
      return `🚨 **Critical Threat Intelligence Advisory: What Engineering & Security Leaders Need to Know**

Cybersecurity threat actors are actively exploiting unpatched gateway endpoints. Here is an actionable breakdown from our intelligence team:

**Key Takeaways:**
🔹 **Vulnerability Profile:** Remote Code Execution leading to privilege escalation.
🔹 **Threat Level:** Critical (Active scanning observed globally).
🔹 **Immediate Action Required:** Patch perimeter interfaces and verify authentication logs.

💡 **Why this matters for CISOs and IT Directors:**
Legacy rule-based detection often misses multi-stage payloads. Ensuring real-time behavioral monitoring and strict zero-trust segmentation is critical to minimize blast radius.

What proactive measures is your team taking today? Let’s discuss in the comments below.

#CyberSecurity #CISO #ThreatIntelligence #DevSecOps #RiskManagement`;

    case "press_release":
      return `# PUBLIC SECURITY ADVISORY & STATEMENT
**FOR IMMEDIATE RELEASE**  
**Date:** ${dateStr}  
**Contact:** media-security@transmute.intel  

---

### Statement on Recent Threat Landscape Findings

Transmute Security Labs has identified a sophisticated cybersecurity campaign impacting public internet infrastructure. Our team has collaborated with CERT authorities to provide defensive countermeasures.

**Summary for Customers and Partners:**
- **Customer Data Protection:** No direct compromise of customer data stores has occurred.
- **Defensive Measures:** Automated mitigations and security rules have been pushed to all protected networks.
- **Recommended User Steps:** Users and administrators should ensure multi-factor authentication (MFA) remains active across all company portals.

We remain committed to maintaining the highest security standards and open communication with the community.`;

    case "slide_deck":
      return `# Presentation Slide Outline: Threat Response & Strategy

---
### Slide 1: Executive Overview
- **Title:** Incident Briefing & Threat Defense Strategy
- **Key Points:**
  - Brief summary of detected threat vectors
  - Immediate defensive response posture
  - Business impact assessment: Minimal operational interruption
- *Speaker Note:* Welcome stakeholders; set reassuring tone highlighting rapid containment.

---
### Slide 2: Threat Landscape & Vector Analysis
- **Title:** Anatomy of the Exploit
- **Key Points:**
  - Initial entry point via public endpoints
  - Multi-stage payload evasion techniques
  - MITRE ATT&CK Matrix mapping (T1190, T1059)
- *Speaker Note:* Walk through attack progression diagram from left to right.

---
### Slide 3: Roadmap to Zero-Trust Hardening
- **Title:** Corrective Action & Next Steps
- **Key Points:**
  - Enhanced network micro-segmentation
  - Autonomous SOC monitoring upgrade
  - Comprehensive staff awareness refresh
- *Speaker Note:* Present budget and timeline estimates for recommended security tooling.`;

    case "video_script":
      return `# Video Narration Script: Cyber Threat Briefing

**Runtime:** 01:30  
**Tone:** ${params.tone}  
**Language:** ${params.language}  

---

**[SCENE 1: 00:00 - 00:15]**  
- **Visual Cue:** Dramatic dark cyber map visual with flashing alerts over global server nodes.  
- **Narrator (VO):** "Within the last 24 hours, security telemetry detected an active exploitation wave targeting enterprise infrastructure. Here is your situational briefing."

---

**[SCENE 2: 00:15 - 00:45]**  
- **Visual Cue:** Animated breakdown of the exploit payload bypassing firewalls and isolating endpoints.  
- **Narrator (VO):** "Attackers utilized chained vulnerabilities to attempt administrative takeover. However, automated detection mechanisms quarantined the affected subnets before lateral movement could occur."

---

**[SCENE 3: 00:45 - 01:15]**  
- **Visual Cue:** 3 bold checkmarks on screen with clear actionable mitigation steps.  
- **Narrator (VO):** "Your immediate priorities: block known malicious C2 IP addresses, inspect gateway authorization logs, and apply the latest emergency vendor patch."

---

**[SCENE 4: 01:15 - 01:30]**  
- **Visual Cue:** Transmute Intelligence logo, link to security portal, and support QR code.  
- **Narrator (VO):** "For full indicators of compromise and detailed remediation scripts, visit the link below. Stay vigilant."`;

    case "playbook":
      return `# Incident Response Remediation Playbook
**Playbook Code:** PB-SEC-09  
**Category:** Emergency Containment & Recovery  

---

## Stage 1: Identification & Verification
1. Verify alert telemetry from SIEM/EDR consoles.
2. Cross-reference source IP against threat intelligence feed (\`198.51.100.42\`).
3. Identify all hosts that initiated outbound sessions to the flagged domain within the past 48 hours.

## Stage 2: Immediate Containment
- **Network Isolation:** Apply VLAN quarantine rule \`QUARANTINE_TIER_1\` to affected virtual machines.
- **Firewall Rule Injection:**
  \`\`\`bash
  iptables -A INPUT -s 198.51.100.42 -j DROP
  iptables -A OUTPUT -d 198.51.100.42 -j DROP
  \`\`\`
- **Session Revocation:** Terminate all active OAuth and JWT sessions for impacted services.

## Stage 3: Eradication & Recovery
1. Re-image compromised nodes using golden baseline templates.
2. Rotate API keys and service principal credentials.
3. Validate integrity checks using hash baseline:
   \`\`\`bash
   sha256sum -c /etc/security/baseline_hashes.sha256
   \`\`\`

## Stage 4: Post-Incident Review
- Compile timeline report within 48 hours.
- Update internal detection signatures and firewall baseline.`;

    default:
      return `# Transmute Deliverable
**Type:** ${id}  
**Audience:** ${params.audienceCategory}  
**Tone:** ${params.tone}  

Generated analysis based on provided intelligence sources.`;
  }
}

export async function regenerateDeliverable(
  id: OutputTypeId,
  sourceText: string,
  params: GenerationParams,
  refinement: string
): Promise<string> {
  const base = await generateDeliverable(id, sourceText, params);
  return `${base}\n\n---\n*Updated with refinement directive: "${refinement}"*`;
}

File: preview_pipeline/mock.py
from typing import List, Dict, Any
from .types import (
    MultiPreviewResult, 
    PlatformPreview, 
    SensitiveDataFlag,
    OutputType,
    LinkedInPreviewContent,
    SocialThreadPreviewContent,
    AdvisoryPreviewContent,
    ExecSummaryPreviewContent,
    IncidentReportPreviewContent,
    PressReleasePreviewContent,
    SlideDeckPreviewContent,
    VideoScriptPreviewContent,
    PlaybookPreviewContent,
)

from .renderers import RENDERERS
from .proofchecker import scan_and_redact

MOCK_STRUCTURED = {
    OutputType.LINKEDIN_POST: LinkedInPreviewContent(
        hook="🚨 If your org runs BankShield middleware, you need to read this immediately.",
        threat_context="The ShadowGate Collective is actively exploiting CVE-2026-41822 (CVSS 9.1) in BankShield v8.x controllers, compromising 3,200+ nodes across 14 regional networks. Initial ingress via 10.14.2.1.",
        key_insights=[
            "Legacy rule-based detection misses multi-stage payloads — behavioral monitoring is now non-negotiable.",
            "Credential harvesting occurred within 6 minutes of initial access — zero-trust segmentation limited blast radius.",
            "Vendor patch available but 67% of regional nodes remain unpatched due to change-control bottlenecks.",
        ],
        actionable_takeaways=[
            "Immediately isolate all external BankShield management interfaces and block inbound on port 4433.",
            "Force global credential revocation and enforce phishing-resistant MFA on all controller consoles.",
            "Ingest ShadowGate IOCs into SIEM/EDR and deploy emergency vendor hotfix within 4 hours.",
        ],
        discussion_prompt="What's your current protocol for third-party middleware patch verification across remote assets?",
        hashtags=["#CyberSecurity", "#ThreatIntel", "#CISO", "#SecOps", "#InfoSec", "#DevSecOps"],
        citations_used=["[^src-1]", "[^src-2]"]
    ),
    OutputType.SOCIAL_THREAD: SocialThreadPreviewContent(
        hook_tweet="1/5 🚨 BREAKING: ShadowGate Collective exploiting CVE-2026-41822 (CVSS 9.1) in BankShield middleware. 3,200+ controllers compromised across 14 regions. Thread 🧵👇",
        exploit_tweet="2/5 ⚡ EXPLOIT CHAIN: Unauthenticated RCE via deserialization flaw → credential dump (T1003) → lateral movement via Remote Services (T1021). Ingress IP: 10.14.2.1 staging creds on internal switches.",
        ioc_tweet="3/5 🔍 KEY IOCs: CVE-2026-41822 | IP 10.14.2.1 (ingress/lateral) | Actor: ShadowGate Collective | Target: BankShield v8.x controllers. Check SIEM for anomalous egress from controller subnets NOW.",
        mitigation_tweet="4/5 🛡️ MITIGATE NOW: 1️⃣ Isolate controller mgmt ports 2️⃣ Rotate ALL admin creds + enforce FIDO2 MFA 3️⃣ Deploy vendor hotfix + monitor for anomalous process exec on endpoints.",
        wrapup_tweet="5/5 🔗 Full advisory + IOCs + SIEM rules submitted to CERT. Retweet to warn peers 🔁 Bookmark for SecOps runbook 🔖 #CyberSecurity #ThreatIntel #ZeroDay",
        all_tweets=[
            "1/5 🚨 BREAKING: ShadowGate Collective exploiting CVE-2026-41822 (CVSS 9.1) in BankShield middleware. 3,200+ controllers compromised across 14 regions. Thread 🧵👇",
            "2/5 ⚡ EXPLOIT CHAIN: Unauthenticated RCE via deserialization flaw → credential dump (T1003) → lateral movement via Remote Services (T1021). Ingress IP: 10.14.2.1 staging creds on internal switches.",
            "3/5 🔍 KEY IOCs: CVE-2026-41822 | IP 10.14.2.1 (ingress/lateral) | Actor: ShadowGate Collective | Target: BankShield v8.x controllers. Check SIEM for anomalous egress from controller subnets NOW.",
            "4/5 🛡️ MITIGATE NOW: 1️⃣ Isolate controller mgmt ports 2️⃣ Rotate ALL admin creds + enforce FIDO2 MFA 3️⃣ Deploy vendor hotfix + monitor for anomalous process exec on endpoints.",
            "5/5 🔗 Full advisory + IOCs + SIEM rules submitted to CERT. Retweet to warn peers 🔁 Bookmark for SecOps runbook 🔖 #CyberSecurity #ThreatIntel #ZeroDay",
        ],
        citations_used=["[^src-1]", "[^src-2]"]
    ),
    OutputType.ADVISORY: AdvisoryPreviewContent(
        tl_protocol="TLP:AMBER+STRICT",
        severity="CRITICAL",
        cvss_score=9.1,
        cve_ids=["CVE-2026-41822"],
        threat_actor="ShadowGate Collective",
        affected_systems=["BankShield middleware v8.x", "BankShield Controller OS v8.0-8.4"],
        executive_summary="Active exploitation of CVE-2026-41822 by ShadowGate Collective has compromised 3,200+ BankShield controllers across 14 regional networks. Credential harvesting and lateral movement confirmed. Immediate isolation and patching required.",
        technical_analysis="Attackers exploited unauthenticated deserialization (CVE-2026-41822) in BankShield ingestion endpoint for initial access (T1190). Post-exploitation: OS credential dumping via LSASS (T1003.001), lateral movement through SMB/WinRM (T1021.002/T1021.006), and C2 beaconing over HTTPS. No persistence on database layer detected.",
        iocs=[
            {"type": "CVE", "indicator": "CVE-2026-41822", "context": "Root exploit vector - unauthenticated RCE", "action": "patch"},
            {"type": "IPv4", "indicator": "10.14.2.1", "context": "Ingress & lateral staging node", "action": "block"},
            {"type": "Domain", "indicator": "telemetry-sync-auth.net", "context": "C2 exfiltration endpoint", "action": "block"},
            {"type": "SHA-256", "indicator": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855", "context": "Stager binary hash", "action": "monitor"},
            {"type": "Actor", "indicator": "ShadowGate Collective", "context": "Primary campaign attribution", "action": "hunt"},
        ],
        mitigations=[
            "IMMEDIATE: Network isolation of all BankShield controller nodes; block inbound 10.14.2.1 at perimeter.",
            "IMMEDIATE: Global credential revocation + session termination for all admin accounts; enforce FIDO2 MFA.",
            "URGENT: Deploy vendor hotfix for CVE-2026-41822 across all regional clusters within 4 hours.",
            "SHORT-TERM: Update SIEM correlation rules for anomalous egress from controller subnets; enable 14-day forensic logging.",
            "STRATEGIC: Implement zero-trust segmentation for all third-party middleware; mandate patch SLAs in vendor contracts.",
        ],
        cert_reporting="Report confirmed indicators to National CERT: incident-response@cert-in.org.in",
        citations_used=["[^src-1]", "[^src-2]", "[^src-3]"]
    ),
    OutputType.EXEC_SUMMARY: ExecSummaryPreviewContent(
        bluf="ShadowGate Collective exploited CVE-2026-41822 in BankShield middleware, compromising 3,200+ controllers across 14 regions; SOC contained perimeter ingress, zero lateral expansion to core transaction pipelines.",
        situation="BankShield middleware v8.x controllers across regional networks exposed via unpatched deserialization vulnerability (CVE-2026-41822, CVSS 9.1). Threat actor ShadowGate Collective achieved initial access via 10.14.2.1.",
        complication="Operational exposure: perimeter switches quarantined to prevent systemic outage; core pipelines online under heightened monitoring. Regulatory: 6-hour mandatory disclosure initiated per central banking/CERT guidelines. Brand/legal: no customer deposit tampering detected.",
        solution="Network isolation applied to affected endpoints; global admin credentials rotated + MFA enforced; digital forensics team deployed for host telemetry preservation; vendor hotfix deployment in progress.",
        strategic_recommendations=[
            "Approve emergency vendor remediation budget for expedited patching + independent code audit.",
            "Authorize coordinated public disclosure holding statement with legal/cyber counsel.",
            "Mandate zero-trust segmentation for all third-party middleware in procurement policy.",
            "Invest in autonomous EDR rollout to reduce dwell time by 60% across controller fleet.",
        ],
        citations_used=["[^src-1]", "[^src-2]"]
    ),
    OutputType.INCIDENT_REPORT: IncidentReportPreviewContent(
        incident_id="INC-2026-9812",
        status="CONTAINED",
        severity="Tier 1 High",
        timeline=[
            {"time": "08:14:22", "event": "Initial anomalous ingress traffic flagged from foreign subnet 10.14.2.1"},
            {"time": "08:21:05", "event": "Privilege escalation alert triggered exploiting CVE-2026-41822 on core switches"},
            {"time": "08:35:00", "event": "SOC initiated perimeter containment and IP blocklist push"},
            {"time": "09:10:14", "event": "Host isolation completed; zero persistence mechanisms found on database layer"},
        ],
        root_cause="CVE-2026-41822 deserialization flaw in BankShield v8.x ingestion endpoint allowed unauthenticated RCE. Attackers dumped admin credentials via LSASS and moved laterally via SMB.",
        blast_radius=[
            "3,200+ BankShield controllers across 14 regional networks",
            "Administrative credentials for controller management consoles",
            "Internal switch management interfaces (no database layer persistence)",
        ],
        corrective_actions=[
            {"action": "Ingress point isolated and firewall blocklists enforced", "status": "complete", "owner": "NetSec"},
            {"action": "Administrative credentials revoked and rotated", "status": "complete", "owner": "IAM"},
            {"action": "14-day continuous telemetry logging activated", "status": "complete", "owner": "SOC"},
            {"action": "Vendor hotfix deployment across regional clusters", "status": "in-progress", "owner": "Infra"},
        ],
        citations_used=["[^src-1]", "[^src-2]"]
    ),
    OutputType.PRESS_RELEASE: PressReleasePreviewContent(
        dateline="NEW DELHI — October 12, 2026",
        headline="National Cyber Agency Confirms Proactive Containment of Banking Infrastructure Security Event",
        customer_impact="Consumer accounts and customer data repositories remain secure and uncompromised. No evidence of deposit tampering or balance alteration.",
        proactive_measures=[
            "Security patches and firewall blocklists deployed across all affected regional nodes within 2 hours of detection.",
            "Automated defensive protocols neutralized unauthorized ingress from external endpoints.",
            "Continuous telemetry monitoring maintained in coordination with national cyber defense agencies.",
            "Full IOC set and SIEM detection rules shared with CERT coordination bodies.",
        ],
        user_guidance=[
            "Ensure multi-factor authentication remains active on all banking portals.",
            "Report suspicious account activity to your bank's official fraud helpline.",
            "Keep banking apps updated to latest versions.",
        ],
        media_contact="press-office@cert-in.org.in | +91-11-XXXX-XXXX",
        citations_used=["[^src-1]"]
    ),
    OutputType.SLIDE_DECK: SlideDeckPreviewContent(
        slides=[
            {"title": "Executive Overview", "type": "TITLE_SLIDE", "key_points": ["Incident Briefing & Threat Defense Strategy", "ShadowGate Collective | CVE-2026-41822 | BankShield Middleware", "Rapid containment confirmed — zero core pipeline impact"], "speaker_notes": "Welcome stakeholders; set reassuring tone highlighting rapid containment."},
            {"title": "Anatomy of the Exploit", "type": "TWO_COLUMN", "key_points": ["LEFT: Technical payload & CVE-2026-41822 deserialization chain", "RIGHT: MITRE ATT&CK — T1190, T1003, T1021, T1071", "Perimeter bypass telemetry & credential staging evidence"], "speaker_notes": "Walk through attack progression left-to-right; emphasize MITRE mapping."},
            {"title": "Remediation Timeline & Hardening", "type": "TIMELINE", "key_points": ["Phase 1 (0-4hrs): Isolation, credential rotation, blocklists", "Phase 2 (4-24hrs): Vendor hotfix deployment, forensic imaging", "Phase 3 (24-72hrs): Zero-trust segmentation, detection rule updates", "Phase 4 (7d): Post-incident review, vendor SLA renegotiation"], "speaker_notes": "Present budget/timeline estimates; highlight Phase 1 complete."},
            {"title": "Strategic Recommendations", "type": "CONCLUSION", "key_points": ["Autonomous EDR rollout — reduce dwell time 60%", "Zero-trust middleware segmentation — procurement mandate", "Vendor patch SLAs in contracts — legal enforcement", "Board-level cyber risk quantification — quarterly review"], "speaker_notes": "Request executive sign-off on 3 budget items."},
        ],
        citations_used=["[^src-1]", "[^src-2]"]
    ),
    OutputType.VIDEO_SCRIPT: VideoScriptPreviewContent(
        runtime_seconds=90,
        scenes=[
            {"scene": "Scene 1 (0:00 - 0:15): Threat Alert Hook", "visual": "Global threat map animation with flashing alert nodes over 14 regional networks", "narrator": "Security telemetry has detected an active campaign targeting enterprise banking infrastructure. ShadowGate Collective is exploiting a critical vulnerability right now. Here is your situational briefing."},
            {"scene": "Scene 2 (0:15 - 0:45): Technical Breakdown", "visual": "Exploit sequence animation: deserialization RCE → credential dump → lateral movement via SMB", "narrator": "Attackers weaponized CVE-2026-41822, an unauthenticated deserialization flaw in BankShield middleware. They achieved remote code execution, dumped administrative credentials, and moved laterally across 3,200 controllers in 14 regions."},
            {"scene": "Scene 3 (0:45 - 1:15): Defense Directives", "visual": "Three bold checkmarks: Isolate Controllers | Rotate Credentials + MFA | Deploy Hotfix & Monitor", "narrator": "Your immediate priorities: First, isolate all BankShield management interfaces and block the ingress IP. Second, revoke every administrative credential and enforce phishing-resistant MFA. Third, deploy the emergency vendor patch and ingest IOCs into your SIEM."},
            {"scene": "Scene 4 (1:15 - 1:30): Conclusion & Resources", "visual": "Transmute Intelligence logo, security portal URL, CERT contact, QR code", "narrator": "Full indicators of compromise, SIEM rules, and remediation scripts are available at the link below. Report confirmed activity to your national CERT. Stay vigilant."},
        ],
        citations_used=["[^src-1]", "[^src-2]"]
    ),
    OutputType.PLAYBOOK: PlaybookPreviewContent(
        playbook_code="PB-SEC-09",
        stages=[
            {"stage": 1, "title": "Identification & Verification", "steps": ["Query SIEM/EDR for CVE-2026-41822 exploitation signatures", "Cross-reference source IP 10.14.2.1 against threat intel feeds", "Identify all hosts with outbound sessions to telemetry-sync-auth.net in past 48h", "Validate BankShield controller version inventory across regions"], "commands": ["grep -r 'CVE-2026-41822' /var/log/siem/", "ioc-check --ip 10.14.2.1 --feed all", "netflow-query --dst-ip 10.14.2.1 --since 48h"]},
            {"stage": 2, "title": "Immediate Containment", "steps": ["Apply VLAN quarantine rule QUARANTINE_TIER_1 to affected controller VMs", "Inject perimeter firewall block for 10.14.2.1 and telemetry-sync-auth.net", "Terminate all active OAuth/JWT sessions for BankShield management consoles", "Disable external access to controller management ports (4433, 8443)"], "commands": ["iptables -A INPUT -s 10.14.2.1 -j DROP", "iptables -A OUTPUT -d 10.14.2.1 -j DROP", "firewall-cmd --permanent --add-rich-rule='rule family=ipv4 source address=10.14.2.1 drop'", "kubectl label nodes bankshield-controller quarantine=true"]},
            {"stage": 3, "title": "Eradication & Recovery", "steps": ["Re-image compromised nodes using golden baseline templates v8.4.1+", "Rotate all API keys, service principals, and database credentials", "Validate integrity using hash baseline: sha256sum -c /etc/security/baseline_hashes.sha256", "Deploy vendor hotfix for CVE-2026-41822 across all regional clusters"], "commands": ["ansible-playbook reimage-controllers.yml --extra-vars 'version=8.4.1'", "vault rotate --path secret/bankshield/*", "sha256sum -c /etc/security/baseline_hashes.sha256", "yum update -y bankshield-middleware-8.4.1"]},
            {"stage": 4, "title": "Post-Incident Auditing", "steps": ["Compile timeline report within 48 hours for CERT submission", "Update internal detection signatures for ShadowGate TTPs", "Review dwell time metrics and logging gaps across controller fleet", "Renegotiate vendor patch SLA contracts with mandatory 24hr critical patch window"], "commands": ["dfir-timeline --incident INC-2026-9812 --output report.pdf", "sigma-rule-gen --actor ShadowGate --output rules/", "log-audit --fleet bankshield --since 30d"]},
        ],
        citations_used=["[^src-1]", "[^src-2]", "[^src-3]"]
    ),
}

def get_mock_previews(content_md: str, selected_outputs: List[str], is_organization: bool) -> MultiPreviewResult:
    previews = {}
    
    for platform in selected_outputs:
        if platform not in MOCK_STRUCTURED:
            continue
            
        structured = MOCK_STRUCTURED[platform]
        renderer = RENDERERS.get(platform)
        draft_content = renderer(structured.model_dump()) if renderer else str(structured)
        citations = structured.citations_used
        
        flags = []
        if is_organization:
            draft_content, flags = scan_and_redact(draft_content)
        
        previews[platform] = PlatformPreview(
            platform_key=platform,
            display_name=platform.replace("_", " ").title(),
            draft_title=f"{platform.replace('_', ' ').title()} Preview",
            draft_content=draft_content,
            structured_content=structured,
            citations_used=citations,
            sensitive_flags=flags
        )
        
    return MultiPreviewResult(
        is_organization=is_organization,
        previews=previews,
        source_summary="Mock threat intelligence: ShadowGate Collective exploiting CVE-2026-41822 in BankShield middleware across 14 regional networks.",
        extracted_facts=[
            "ShadowGate Collective exploiting CVE-2026-41822 (CVSS 9.1)",
            "3,200+ BankShield controllers compromised across 14 regions",
            "Ingress IP 10.14.2.1 used for credential staging",
            "Admin credentials harvested via LSASS dump",
            "Lateral movement via SMB/WinRM",
            "Zero persistence on database layer",
            "Vendor hotfix available for CVE-2026-41822",
        ],
        metadata_anchors={
            "threat_actor": "ShadowGate Collective",
            "cve": "CVE-2026-41822",
            "cvss": 9.1,
            "target": "BankShield middleware v8.x",
            "ingress_ip": "10.14.2.1",
            "c2_domain": "telemetry-sync-auth.net",
        }
    )